src/main/java/com/github/ferstl/maven/pomenforcers/PedanticDependencyConfigurationEnforcer.java

```diff
@@ -143,26 +186,4 @@ private Collection<DependencyModel> searchForDependencies(Predicate<DependencyMo
     List<DependencyModel> dependencies = getProjectModel().getDependencies();
     return dependencies.stream().filter(predicate).collect(Collectors.toList());
   }
-
-  private enum DependencyPredicate implements Predicate<DependencyModel> {
-    WITH_VERSION {
-      @Override
-      public boolean test(DependencyModel input) {
-        return input.getVersion() != null;
-      }
-    },
-    WITH_PROJECT_VERSION {
-      @Override
-      public boolean test(DependencyModel input) {
-        return !"${project.version}".equals(input.getVersion())
-            && !"${version}".equals(input.getVersion());
-      }
-    },
-    WITH_EXCLUSION {
-      @Override
-      public boolean test(DependencyModel input) {
-        return !input.getExclusions().isEmpty();
-      }
-    }
-  }
```

Comment:
Unfortunately had to get rid of the `enum` setup since I had to `test` against a Collection which isn't part of the enum itself.

Reply:
This is OK. The enum comes from version 1.x where Java 7 was still supported.